Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -158,7 +158,29 @@ void BTDiagnostics::DerivedInitData ()
     // j >= i / gamma / (1+beta) * dt_snapshot / dt_boosted_frame
     const int final_snapshot_starting_step = static_cast<int>(std::ceil(final_snapshot_iteration / warpx.gamma_boost / (1._rt+warpx.beta_boost) * m_dt_snapshots_lab / dt_boosted_frame));
     const int final_snapshot_fill_iteration = final_snapshot_starting_step + num_buffers * m_buffer_size - 1;
-    if (final_snapshot_fill_iteration > warpx.maxStep()) {
+    const amrex::Real final_snapshot_fill_time = final_snapshot_fill_iteration * dt_boosted_frame;
+    if (warpx.compute_max_step_from_btd) {
+        if (final_snapshot_fill_iteration > warpx.maxStep()) {
+            warpx.updateMaxStep(final_snapshot_fill_iteration);
+            amrex::Print()<<"max_step insufficient to fill all BTD snapshots. Automatically increased to: "
+                << final_snapshot_fill_iteration << std::endl;
+
+        } else if (final_snapshot_fill_time > warpx.stopTime()) {
+            warpx.updateStopTime(final_snapshot_fill_time);
+            amrex::Print()<<"stop_time insufficient to fill all BTD snapshots. Automatically increased to: "
+                << final_snapshot_fill_time << std::endl;
+
+        } else if (warpx.maxStep() == std::numeric_limits<int>::max() && warpx.stopTime() == std::numeric_limits<amrex::Real>::max()) {
+            amrex::Print()<<"max_step unspecified.  Setting to "
+                <<final_snapshot_fill_iteration<< " to fill all BTD snapshots." << std::endl;
+            warpx.updateMaxStep(final_snapshot_fill_iteration);
+            amrex::Print()<<"stop_time unspecified.  Setting to "
+                <<final_snapshot_fill_time<< " to fill all BTD snapshots." << std::endl;
+            warpx.updateMaxStep(final_snapshot_fill_time);
+
+        }
```

Comment:
Is this the case where `zmax_plasma_to_compute_time_step` is specified? Or is it possible that neither of the three are specified by the user? If this corresponds to the case where `zmax_plasma_to_compute_time_step` is specified, when does the computation of `max_step` (or `stop_time`) based on `zmax_plasma_to_compute_time_step` occur, before or after this function is called? If before, I would assume that we actually do not end up in this `else if` block, but then I'm lost about which case this covers. Might be useful to clarify this offline.

Reply:
This (or rather a similar block in a newer version) is the case when none of ``max_step``, ``stop_time``, or ``zmax_plasma_to_compute_time_step`` are specified but ``compute_max_step_from_btd=1``.